File: trunk/quests/qeynos/Zamel.pl
#############
#Quest Name: Finding Lucie Elron (1/3)
#Author: BWStripes
#NPC's Involved: 2
#Items involved: 1
#############
###NPC 1
#Name: Zamel
#Race 1 (Human), Texture of 0, Size 6, gender of 0
#Location X,Y,Z: -395.0, 149.0, 3.7 in South Qeynos
#Level: 10
#Type: Quest NPC
#Reward:  Random between: -
# 10018:Hematite
# 10320:Golden Ear Stud
#############
###Item 1
#Name: A Small Tattered Dress
#ID: 13713
###

#Description: The point of this quest is to locate the child. Sadly, she has been eaten by a viscious (or is that viscous?) gel cube from the Qeynos Sewers.
#You randomly find the remains of her bloodied dress on Cuburt. This was a boon to low level int casters as the reward was randomly (I recall one in 5 ish) a +3 int earring.
#Personal note: EQEmu spawn/drop rate on the dress from Cuburt is awful, compared to live, so I have upped the "good" reward to 35% chance.
#Other NPCs that you can hand the dress into: Guard Elron, North Qeynos; Endric, South Qeynos

#Quest for Zamel in South Qeynos
sub EVENT_SAY {
  if($text=~/Hail/i){
    quest::say("Greetings, good $name.  My name is Zamel and I have the much coveted duty of pulling sludge up out of the pools and [aqueducts] of Qeynos..");
    }
  if($text=~/what aqueducts/i){
    quest::say("Aye, the aqueducts are a horrible place.  I have heard rumors of horrible [beasts] who live under the city in the water and just the other day, a child playing in the water just up and disappeared!");
  }
  if($text=~/what beasts/i){
    quest::say("I know nothing of what or where they may be, just rumors, friend.  However, should you foolishly choose to enter the waters to brave the possibilities and return with some evidence of the [child's] fate, I would be most appreciative.");
  }
  if($text=~/what child/i){
    quest::say("The daughter of one of our own guardsmen is missing.  He has offered me a reward for any information about her disappearance.  So, if you find out anything, let me know and I will cut you in on the reward.");
  }
}

sub EVENT_ITEM {
  if(plugin::check_handin(\%itemcount, 13713 => 1)){
    quest::say("Yes,  I do believe this was the child's dress.  Here, take this as your share of the reward.  Her father will be heartbroken..");
    my $ZamelReward = int(rand(100));
    if ($ZamelReward<=35){
      quest::summonitem("10320","1");
    }
    else {
      quest::summonitem("10018","1");
    }
    quest::givecash("0","4","0","0");
    quest::faction(36,5); #Guards of Qeynos
    quest::faction(9,5); #Antonius Bayle
    quest::faction(53,-1); #Corrupt Guards of Qeynos
    quest::faction(33,-1); #Circle Of Unseen Hands
    quest::faction(217,5); #Merchants of Qeynos
    quest::exp("100");
  }
}

#End of updated file Zamel.pl

File: trunk/quests/qeynos/Zamel.lua
function event_spawn(e)
	eq.set_timer("help",300000);
end

function event_timer(e)
	local endicRandom = 0;
	if(e.timer == "help") then
		endicRandom = math.random(100);
		if(endicRandom < 16) then
			e.self:Say("Clean the [aqueducts] they tell me. Peh, they don't pay me enough to climb into that filth!");
		end		
	end
end

function event_say(e)
	if(e.message:findi("hail")) then
		e.self:Say(string.format("Greetings, good %s.  My name is Zamel and I have the much coveted duty of pulling sludge up out of the pools and [aqueducts] of Qeynos..",e.other:GetName()));
	elseif(e.message:findi("aqueducts")) then
		e.self:Say("Aye, the aqueducts are a horrible place.  I have heard rumors of horrible [beasts] who live under the city in the water and just the other day, a child playing in the water just up and disappeared!");
	elseif(e.message:findi("beasts")) then
		e.self:Say("I know nothing of what or where they may be, just rumors, friend.  However, should you foolishly choose to enter the waters to brave the possibilities and return with some evidence of the [child's] fate, I would be most appreciative.");
	elseif(e.message:findi("child")) then
		e.self:Say("The daughter of one of our own guardsmen is missing.  He has offered me a reward for any information about her disappearance.  So, if you find out anything, let me know and I will cut you in on the reward.");
	end
end

function event_trade(e)
	local item_lib =require("items");
	if(item_lib.check_turn_in(e.trade, {item1 = 13713})) then
		e.self:Say("Yes,  I do believe this was the child's dress.  Here, take this as your share of the reward.  Her father will be heartbroken..");
		e.other:SummonItem(eq.ChooseRandom(10018,10018,10320));
		e.other:Ding();
		e.other:Faction(36,5,0);
		e.other:Faction(9,5,0);
		e.other:Faction(53,-5,0);
		e.other:Faction(33,-5,0);
		e.other:Faction(217,5,0);
		e.other:GiveCash(0,4,0,0);
		e.other:AddEXP(100);
	end
	item_lib.return_items(e.self, e.other, e.trade)
end
